src/datakit_io.ml

```diff
@@ -116,26 +117,93 @@ module FS = struct
   let protect f x = Lwt.catch (fun () -> f x) protect_unix_exn
   let safe f x = Lwt.catch (fun () -> f x) ignore_enoent
 
-  let remove_file f = safe Lwt_unix.unlink f
-
   let mkdir dirname =
     let rec aux dir =
       if Sys.file_exists dir && Sys.is_directory dir then Lwt.return_unit
       else (
         let clear =
           if Sys.file_exists dir then (
-            Log.debug (fun f -> f "%s already exists but is a file, removing." dir);
-            remove_file dir;
+            Log.debug (fun l ->
+                l "%s already exists but is a file, removing." dir);
+            safe Lwt_unix.unlink dir
           ) else
             Lwt.return_unit
         in
         clear >>= fun () ->
         aux (Filename.dirname dir) >>= fun () ->
-        Log.debug (fun f -> f "mkdir %s" dir);
+        Log.debug (fun l -> l "mkdir %s" dir);
         protect (Lwt_unix.mkdir dir) 0o755;
       ) in
     Lwt_pool.use mkdir_pool (fun () -> aux dirname)
 
+  let file_exists f =
+    Lwt.catch (fun () -> Lwt_unix.file_exists f) (function
+        (* See https://github.com/ocsigen/lwt/issues/316 *)
```

Comment:
This issue is now fixed in Lwt>=2.7.1 -- update constraint and remove this workaround?